kernel/src/grant.rs

```diff
@@ -102,7 +102,7 @@ impl Allocator {
 
                         // We use `ptr::write` to avoid `Drop`ping the uninitialized memory in
                         // case `T` implements the `Drop` trait.
-                        write(ptr.as_ptr(), data);
+                        write(ptr.as_ptr(), T::default());
```

Comment:
I'm pretty uncertain why this change would actually solve the "grants temporarily take up a ton of stack space" problem. `T::default()` is still going to return a T, which will be its full size and on the stack. Is passing it into the `alloc_unowned()` function taking double the size of T on the stack or something?

Reply:
Per a slack discussion with bradjc, this change greatly increases the possible size of IPC grants before a kernel stack overflow occurs (previously 364 bytes increases to 2364 bytes). I'm guessing some compiler optimizations are making the magic happen here.

Reply:
Right, presumably the compiler can better guess to just allocate the space for the default value directly at `ptr.as_ptr()` as opposed to on the stack first.